Gems/Atom/RPI/Code/Include/Atom/RPI.Public/BlockCompression.h

```diff
@@ -78,11 +78,13 @@ namespace AZ
                     return extractColor(m_color1);
                     break;
                 case 2:
-                    // Index 2 uses 2/3 of color 0 and 1/3 of color 1
-                    return extractColor(m_color0).Lerp(extractColor(m_color1), (1.0f / 3.0f));
+                    // Index 2 either uses 2/3 of color 0 and 1/3 of color 1 or 1/2 of color 0 and 1/2 of color 1
+                    return (m_color0 > m_color1) ? extractColor(m_color0).Lerp(extractColor(m_color1), (1.0f / 3.0f))
```

Comment:
is it worth doing the color0 > color1 check at block init rather than for every pixel decode?

Reply:
There is no actual block init to do it at, the struct doesn't get constructed / destroyed, it gets overlaid on top of existing data.

Reply:
The usage in RPIUtils.cpp looks like this:
```
                case AZ::RHI::Format::BC1_UNORM:
                    {
                        auto actualMem = reinterpret_cast<const BC1Block*>(mem);
                        return actualMem[indices.first].GetBlockColor(indices.second).GetElement(componentIndex);
                    }
```